Release/src/utilities/asyncrt_utils.cpp

```diff
@@ -352,14 +352,22 @@ inline size_t count_utf8_to_utf16(const std::string& s)
     const size_t sSize = s.size();
     const char* const sData = s.data();
     size_t result{ sSize };
+
     for (size_t index = 0; index < sSize;)
     {
-        const char c{ sData[index++] };
-        if ((c & BIT8) == 0)
+        if( sData[index] > 0 )
```

Comment:
This transformation doesn't work on platforms where char is unsigned -- e.g., most non-Windows platforms. Clearly this is missing some test coverage if that was missed...

Reply:
You're right, I missed that. I fixed the code, still thinking about testability. This would probably require a separate test binary with different char defaults.

Reply:
I think the tests might be avoiding this with the __GLIBCXX__ bits? (GCC defaults to char being unsigned)

Reply:
`__GLIBCXX__` seems only to be used as a means to determine whether the test code should compare byte-wise (using unsigned char) or via the results of conversion derived from codecvt_utf8_utf16. Where I'm not sure about the reason for this, especially since codecvt_utf8_utf16 is deprecated in C++17 (->https://en.cppreference.com/w/cpp/locale/codecvt_utf8_utf16).

With regard to the test, I was considering whether there is an elegant way to make use of the effects of `-funsigned-char` and `-fsigned-char` with GCC (-> http://gcc.gnu.org/onlinedocs/gcc-4.8.0/gcc/C-Dialect-Options.html) or /J Option in VS (-> https://msdn.microsoft.com/en-us/library/0d294k5z.aspx). Where elegant would mean that we at least avoid building the tests with their dependecies twice (once signed once unsigned).

Reply:
OK, but this would prevent the change to work on unsigned char platforms without changes. Or is this just a marker to ensure unsigned char checks are done before next release?

Reply:
Another thought: We could reinterpret_cast the pointer retrieved from s.data() to signed char (or unsigned whatever fits better to the project) and avoid the platform dependency. From what I see, this should also be fine with regard to strict aliasing rules since we're only working on the buffer via our pointers and are not using any std::string operations.
What do you think?

Reply:
That should be OK (because signed char/unsigned char are character types)